internal/cmd/xmrswap/main.go

```diff
@@ -0,0 +1,1221 @@
+//go:build libsecp256k1
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/config"
+	dcradaptor "decred.org/dcrdex/internal/adaptorsigs/dcr"
+	"decred.org/dcrdex/internal/libsecp256k1"
+	"decred.org/dcrwallet/v4/rpc/client/dcrwallet"
+	dcrwalletjson "decred.org/dcrwallet/v4/rpc/jsonrpc/types"
+	"github.com/agl/ed25519/edwards25519"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/chaincfg/v3"
+	"github.com/decred/dcrd/dcrec"
+	"github.com/decred/dcrd/dcrec/edwards/v2"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/decred/dcrd/rpcclient/v8"
+	"github.com/decred/dcrd/txscript/v4"
+	"github.com/decred/dcrd/txscript/v4/sign"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
+	"github.com/decred/dcrd/wire"
+	"github.com/decred/slog"
+	"github.com/dev-warrior777/go-monero/rpc"
+	"github.com/fatih/color"
+	"github.com/haven-protocol-org/monero-go-utils/base58"
+)
+
+// TODO: Verification at all stages has not been implemented yet.
```

Comment:
This TODO still valid?

Reply:
Yes, we are not inspecting the on-chain txn at all yet.